### PR TITLE
salt: 3006.5 (new cask)

### PR DESCRIPTION
**Pull request by cdalvaro**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online salt` is error-free.
- [x] `brew style --fix salt` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new salt` worked successfully.
  - This check fails because there is a formula in the core repo called `salt`. Although that formula is deprecated and this cask is intended to replace it.
- [x] `brew install --cask salt` worked successfully.
- [x] `brew uninstall --cask salt` worked successfully.

### CODE REVIEW

**Comment by cdalvaro** [comment]
This cask is a replacement for the deprecated `salt` formula.

**Comment by cdalvaro** [comment]
> This cask is a replacement for the deprecated and disabled `salt` formula. Maybe it's time to remove the `salt` formula from the _core_ repository?

I would like to know @SMillerDev's opinion about this.

**Comment by SMillerDev** [comment]
Sounds fine to me, it's been deprecated and disabled for a while already.

File: Casks/s/salt.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                      args: ["-lint", new_plist_file]
+
+      system_command! "mv",
+                      args: [new_plist_file, plist_file],
+                      sudo: true
+      system_command! "chown",
+                      args: ["root:wheel", plist_file],
+                      sudo: true
+    end
+  end
```
Comment (bevanjkay) [inline Casks/s/salt.rb:50]: Can you provide some commentary on why this `postflight` block is necessary, and perhaps add a comment to code explaining it succinctly?

File: Casks/s/salt.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  uninstall pkgutil:   [
+              "com.saltstack.salt",
+            ],
+            launchctl: [
+              "com.saltstack.salt.*",
```
Comment (bevanjkay) [inline Casks/s/salt.rb:56]: I don't believe that globs are supported for `launchctl`.

File: Casks/s/salt.rb (hunk 1 above)
Comment (SMillerDev) [reply Casks/s/salt.rb:50]: I don't think we should use a post block to fix the deficiencies in salt.

Comment (cdalvaro) [reply Casks/s/salt.rb:50]: So, as an alternative, could this be handled as an optional step, disabled by default?

Comment (SMillerDev) [reply Casks/s/salt.rb:50]: No, it should be fixed upstream

Comment (SMillerDev) [reply Casks/s/salt.rb:50]: Maybe make an issue/PR upstream and link there with a caveat?

Comment (cdalvaro) [reply Casks/s/salt.rb:50]: I'll try to improve this PR: https://github.com/saltstack/salt/pull/64924

So, I think neither the modification of the plist files, nor the caveats documentation will be necessary.

Comment (cdalvaro) [reply Casks/s/salt.rb]: `postflight` removed

File: Casks/s/salt.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      replacing the placeholder text with custom information:
+
+      sudo salt-config -i [MINION_ID] -m [SALT_MASTER_HOST]
+
+      See Configure the Salt master and minions for more configuration options:
+      https://docs.saltproject.io/salt/install-guide/en/latest/topics/configure-master-minion.html
+
+      Load services you need:
+
+      sudo launchctl load -w /Library/LaunchDaemons/com.saltstack.salt.api
```
Comment (SMillerDev) [inline Casks/s/salt.rb:48]: Shouldn't this be a `.plist`?

File: Casks/s/salt.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  def caveats
+    <<~CAVEATS
+      After installation, configure the Salt minion ID, and the Salt master location,
+      replacing the placeholder text with custom information:
+
+      sudo salt-config -i [MINION_ID] -m [SALT_MASTER_HOST]
+
+      See Configure the Salt master and minions for more configuration options:
+      https://docs.saltproject.io/salt/install-guide/en/latest/topics/configure-master-minion.html
```
Comment (SMillerDev) [inline Casks/s/salt.rb:44]: Since this is not a homebrew specific instruction I don't think it should be a caveat

File: Casks/s/salt.rb (hunk 3 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb (hunk 4 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  name "Salt"
+  desc "World's fastest, most intelligent and scalable automation engine"
+  homepage "https://saltproject.io/"
+
+  livecheck do
+    url "https://repo.saltproject.io/salt/py3/macos/latest"
+    regex(/salt-(\d+(?:\.\d+)?(?:-\d+)?)-py3-#{arch}\.pkg/)
+  end
+
+  conflicts_with formula: "salt"
```
Comment (bevanjkay) [inline Casks/s/salt.rb:18]: Now that `salt` has been removed from `homebrew-core`, this line needs to be removed here too.

File: Casks/s/salt.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    "/etc/salt",
+  ]
+
+  def caveats
+    <<~CAVEATS
+      After installation, configure the Salt minion ID, and the Salt master location,
+      replacing the placeholder text with custom information:
+
+      sudo salt-config -i [MINION_ID] -m [SALT_MASTER_HOST]
+
```
Comment (SMillerDev) [inline Casks/s/salt.rb:42]: Since this is not specific to the Homebrew install.
```suggestion
```

File: Casks/s/salt.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      Load services you need:
```
Comment (SMillerDev) [inline Casks/s/salt.rb:43]:
```suggestion
      Included services:
```

Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb (hunk 6 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb (hunk 5 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
+cask "salt" do
+  arch arm: "arm64", intel: "x86_64"
+
+  version "3006.5"
+  sha256 arm:   "bac43fcb5ca04679245e1b6510cb060f282c6f1bb15280396214ca671c74ce65",
+         intel: "90c55ae9836d7f3f1e2c6cb930cff60b334393c87482f26b32121c04db2ad5ea"
+
+  url "https://repo.saltproject.io/salt/py3/macos/minor/#{version}/salt-#{version}-py3-#{arch}.pkg"
+  name "Salt"
+  desc "World's fastest, most intelligent and scalable automation engine"
```
Comment (bevanjkay) [inline Casks/s/salt.rb:10]:
```suggestion
  desc "Automation and infastructure management engine"
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/s/salt.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  homepage "https://saltproject.io/"
+
+  livecheck do
+    url "https://repo.saltproject.io/salt/py3/macos/latest"
+    regex(/salt-(\d+(?:\.\d+)?(?:-\d+)?)-py3-#{arch}\.pkg/)
```
Comment (bevanjkay) [inline Casks/s/salt.rb:15]:
```suggestion
    regex(/salt[._-]v?(\d+(?:\.\d+)+)-py3-#{arch}\.pkg/)
```

File: Casks/s/salt.rb (hunk 8 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+            launchctl: [
+              "com.saltstack.salt.api",
+              "com.saltstack.salt.master",
+              "com.saltstack.salt.minion",
+              "com.saltstack.salt.syndic",
+            ]
+
+  zap trash: [
+    "/etc/salt",
+  ]
```
[the remark this answers is not in the document]
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

File: Casks/s/salt.rb (hunk 9 above)
Comment (cdalvaro) [reply Casks/s/salt.rb]: Done!

**Comment by bevanjkay** [comment]
Thanks @cdalvaro 

Using `ci-syntax-only` to skip the token conflict check.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**